pkgs/tools/nix/nixos-render-docs/src/nixos_render_docs/__init__.py

```diff
@@ -1,12 +1,41 @@
 import argparse
 import os
 import sys
+import traceback
+from io import StringIO
+from pprint import pprint
 from typing import Any, Dict
 
 from .md import Converter
 from . import manual
 from . import options
 
+def pretty_print_exc(e: BaseException, *, _desc_text: str = "error") -> None:
+    print(f"\x1b[1;31m{_desc_text}:\x1b[0m", file=sys.stderr)
+    # destructure Exception and RuntimeError specifically so we can show nice
+    # messages for errors that weren't given their own exception type with
+    # a good pretty-printer.
+    if type(e) is Exception or type(e) is RuntimeError:
+        args = e.args
+        if len(args) and isinstance(args[0], str):
+            print("\t", args[0], file=sys.stderr, sep="")
+            args = args[1:]
+        if len(args):
+            print(f"\x1b[1;34mextra info:\x1b[0m", file=sys.stderr)
+            sep = ""
+            for arg in args:
+                print("", file=sys.stderr, end=sep)
+                sep = "\n"
+                buf = StringIO()
+                pprint(arg, stream=buf)
+                for l in buf.getvalue().splitlines():
+                    print("\t", l, file=sys.stderr, sep="")
+    else:
+        print(e)
```

Comment:
This is bit gnarly but when I tried to clean it up, I only got more nesting :woman_shrugging::

```python
match e:
    case Exception() | RuntimeError():
        args = e.args
        match args:
            case [str() as msg, *args]:
                print("\t", msg, file=sys.stderr, sep="")

        if len(args):
            print(f"\x1b[1;34mextra info:\x1b[0m", file=sys.stderr)
            sep = ""
            for arg in args:
                print("", file=sys.stderr, end=sep)
                sep = "\n"
                buf = StringIO()
                pprint(arg, stream=buf)
                print(textwrap.indent(buf.getvalue(), "\t"), file=sys.stderr, end="")
    case _:
        print(e)
```